src/main/kotlin/org/ageseries/libage/sim/thermal/Thermal.kt

```diff
@@ -0,0 +1,317 @@
+    }
+}
+
+/**
+ * Represents a connection between a [ThermalMass] and something else.
+ *
+ * The current implementations involve another ThermalMass and the environment.
+ */
+interface Connection {
+    /**
+     * Compute the heat transfer, in Joules, to do to each side of the connection over [dt] seconds.
+     *
+     * By convention, the returned energy transfers are in argument order; whereas the first is usually the
+     * [ThermalMass], the second may or may not be relevant depending on whether there is a heat sink.
+     */
+    fun transfer(dt: Double): Pair<Double, Double>
+
+    /**
+     * [ThermalMass]es owned by this connection, so the [Simulator] can manage them.
+     */
+    val masses: List<ThermalMass>
+}
+
+class MassConnection(
+    /** One of the connected masses. */
+    val a: ThermalMass,
+    /** The other connected mass. */
+    val b: ThermalMass,
+    /** Thermal parameters of this connection. */
+    val params: ConnectionParameters = ConnectionParameters.DEFAULT,
+): Connection {
+    private var prevFlux: Double = 0.0
+
+    override fun toString() = "<MassConn $a $b $params>"
+
+    /**
+     * Returns the energies to add to [a] and [b] (in that order), which would attempt to equilibriate the connected thermal masses over the given period of time [dt] (in s).
+     *
+     * For this to be stable, [dt] must be held relatively small, since this is a linear approximation to an exponential curve--otherwise, overshoot may be observed.
+     */
+    override fun transfer(dt: Double): Pair<Double, Double> {
+        // Kelvin
+        val deltaT = b.temperature.kelvin - a.temperature.kelvin
+        // W/K
+        val distCondA = a.material.thermalConductivity * params.contactPoint * params.distance
+        val distCondB = b.material.thermalConductivity * (1.0 - params.contactPoint) * params.distance
+        val overallCond = params.area * (distCondA * distCondB * params.conductance).pow(1.0 / 3.0)
+        // W
+        val power = deltaT * overallCond
+        // J
+        val energy = power * dt
+        val critDamp = sqrt((a.mass + b.mass) / overallCond) * dt
+        val dTerm = prevFlux * critDamp * dt
+        prevFlux = power
+        var toA = energy - dTerm
+        var toB = -energy + dTerm
+        if(toA > 0.0) { toA *= params.efficiency }
+        if(toB > 0.0) { toB *= params.efficiency }
+        /*
+        // Detect an inflection point--if the energies would collide, assume they equilibriate.
+        val newTempA = a.temperatureAt(a.energy + energy).kelvin
+        val newTempB = b.temperatureAt(b.energy - energy).kelvin
+        if(deltaT.sign != (newTempB - newTempA).sign) {
+            // Distribute the energies at equilibrium.
+            val totalE = a.energy + b.energy
+            val eqAEnergy = totalE * (a.mass / b.mass) * (a.material.specificHeat / b.material.specificHeat)
+            return eqAEnergy - a.energy
+        }
+        */
```

Comment:
Delete commented code. We have version control for this.

Reply:
This comment block is the only time (for now) this code is going to be checked in; I guess it's history now, so away it goes :)

It's worth looking into getting it working later--I suspect it's just a minor issue.